tests/network/primary_pod_network.go

```diff
@@ -0,0 +1,137 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+	cd "kubevirt.io/kubevirt/tests/containerdisk"
+)
+
+var _ = SIGDescribe("Primary Pod Network", func() {
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		var err error
+		virtClient, err = kubecli.GetKubevirtClient()
+		tests.PanicOnError(err)
+	})
+
+	Describe("Status", func() {
+		AssertReportedIP := func(vmi *v1.VirtualMachineInstance) {
+			By("Getting pod of the VMI")
+			vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+
+			By("Making sure IP reported on the VMI matches the one on the pod")
+			Expect(vmi.Status.Interfaces[0].IP).To(Equal(vmiPod.Status.PodIP), "The address reported in VMI's IP should match the PodIP")
+			Expect(vmi.Status.Interfaces[0].IPs).NotTo(BeEmpty(), "VMI should report a list of its IP addresses")
+			Expect(vmi.Status.Interfaces[0].IPs[0]).To(Equal(vmiPod.Status.PodIP), "The first address reported in VMI's IPs should match the PodIP")
```

Comment:
1. Please drop the text as discussed earlier, it has no added value. The asserted structures are well names already.
2. There is no need for two asserts here, one is enough. Just change the 2nd to a `ContainElement`.

Reply:
1. Done
2. It is needed. The expectation is that we do the same as Kubernetes while reporting IPs. That is, report the first address from `IPs` in `IP` to keep the backwards compatibility.

Reply:
You mean that you want to make sure it is the first element? (i.e. order matters while "ContainElement" ignores it)

Reply:
Correct.

```
    // podIPs holds the IP addresses allocated to the pod. If this field is specified, the 0th entry must
    // match the podIP field. Pods may be allocated at most 1 value for each of IPv4 and IPv6. This list
    // is empty if no IPs have been allocated yet.
```

https://github.com/kubernetes/api/blob/master/core/v1/types.go#L3555